### ISSUE TITLE
GDALGetMetadata returns incorrect WMTS subdatasets

### ISSUE THREAD

**Issue by michkowalczuk** [OSGeo/gdal#10732]
### What is the bug?

`GDALGetMetadata` from C API returns incorrect subdataset names for the following WMTS service:
`https://geoinformation.eisenbahn-bundesamt.de/wmts/dzsf/WMTSCapabilities.xml`
These subdatasets are different from those returned by the `gdalinfo`.

### Steps to reproduce the issue

`GDALGetMetadata(DatasetHandle, "SUBDATASETS");` returns the following subdatasets:
```
'SUBDATASET_1_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/geonetwork/srv/eng/csw?Service=CSW&Request=GetRecordById&Version=2.0.2&id=766f18e4-9b4c-4338-8638-2d78006b0ef0&outputSchema=http://www.isotc211.org/2005/gmd&elementSetName=full,layer=hangrutschungsgefaehrdung_wmts'
'SUBDATASET_1_DESC=Layer Hangrutschungsgefährdung WMTS'
'SUBDATASET_2_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/geonetwork/srv/eng/csw?Service=CSW&Request=GetRecordById&Version=2.0.2&id=766f18e4-9b4c-4338-8638-2d78006b0ef0&outputSchema=http://www.isotc211.org/2005/gmd&elementSetName=full,layer=hochwassergefaehrdung_wmts'
'SUBDATASET_2_DESC=Layer Hochwassergefährdung_HQ100 WMTS'
'SUBDATASET_3_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/geonetwork/srv/eng/csw?Service=CSW&Request=GetRecordById&Version=2.0.2&id=766f18e4-9b4c-4338-8638-2d78006b0ef0&outputSchema=http://www.isotc211.org/2005/gmd&elementSetName=full,layer=potentielle_hangmuren_wmts'
'SUBDATASET_3_DESC=Layer potentielle Hangmuren WMTS'
'SUBDATASET_4_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/geonetwork/srv/eng/csw?Service=CSW&Request=GetRecordById&Version=2.0.2&id=766f18e4-9b4c-4338-8638-2d78006b0ef0&outputSchema=http://www.isotc211.org/2005/gmd&elementSetName=full,layer=potentielle_murgaenge_wmts'
'SUBDATASET_4_DESC=Layer potentielle Murgänge WMTS'
'SUBDATASET_5_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/geonetwork/srv/eng/csw?Service=CSW&Request=GetRecordById&Version=2.0.2&id=766f18e4-9b4c-4338-8638-2d78006b0ef0&outputSchema=http://www.isotc211.org/2005/gmd&elementSetName=full,layer=potentielle_murgaenge_mangmuren_wmts'
'SUBDATASET_5_DESC=Layer potentielle Murgänge und Hangmuren WMTS'
'SUBDATASET_6_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/geonetwork/srv/eng/csw?Service=CSW&Request=GetRecordById&Version=2.0.2&id=766f18e4-9b4c-4338-8638-2d78006b0ef0&outputSchema=http://www.isotc211.org/2005/gmd&elementSetName=full,layer=hangmurengefaehrdung_wmts'
'SUBDATASET_6_DESC=Layer Hangmurengefährdung WMTS'
'SUBDATASET_7_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/geonetwork/srv/eng/csw?Service=CSW&Request=GetRecordById&Version=2.0.2&id=766f18e4-9b4c-4338-8638-2d78006b0ef0&outputSchema=http://www.isotc211.org/2005/gmd&elementSetName=full,layer=murganggefaehrdung_wmts'
'SUBDATASET_7_DESC=Layer Murganggefährdung WMTS'
'SUBDATASET_8_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/geonetwork/srv/eng/csw?Service=CSW&Request=GetRecordById&Version=2.0.2&id=766f18e4-9b4c-4338-8638-2d78006b0ef0&outputSchema=http://www.isotc211.org/2005/gmd&elementSetName=full,layer=mur_und_hangmur_wmts'
'SUBDATASET_8_DESC=Layer Murgang- und Hangmurengefährdung WMTS'
'SUBDATASET_9_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/geonetwork/srv/eng/csw?Service=CSW&Request=GetRecordById&Version=2.0.2&id=766f18e4-9b4c-4338-8638-2d78006b0ef0&outputSchema=http://www.isotc211.org/2005/gmd&elementSetName=full,layer=boeschungsbrandgefaehrdung_agg_wmts'
'SUBDATASET_9_DESC=Layer (Beta) Böschungsbrandgefährdung aggregiert WMTS'
'SUBDATASET_10_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/geonetwork/srv/eng/csw?Service=CSW&Request=GetRecordById&Version=2.0.2&id=766f18e4-9b4c-4338-8638-2d78006b0ef0&outputSchema=http://www.isotc211.org/2005/gmd&elementSetName=full,layer=boeschungsbrandgefaehrdung_wmts'
'SUBDATASET_10_DESC=Layer (Beta) Böschungsbrandgefährdung WMTS'
```

and `gdalinfo "WMTS:https://geoinformation.eisenbahn-bundesamt.de/wmts/dzsf/WMTSCapabilities.xml?"` returns:
```
Subdatasets:
  SUBDATASET_1_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/wmts/dzsf/WMTSCapabilities.xml?,layer=hangrutschungsgefaehrdung_wmts
  SUBDATASET_1_DESC=Layer Hangrutschungsgef├ñhrdung WMTS
  SUBDATASET_2_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/wmts/dzsf/WMTSCapabilities.xml?,layer=hochwassergefaehrdung_wmts
  SUBDATASET_2_DESC=Layer Hochwassergef├ñhrdung_HQ100 WMTS
  SUBDATASET_3_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/wmts/dzsf/WMTSCapabilities.xml?,layer=potentielle_hangmuren_wmts
  SUBDATASET_3_DESC=Layer potentielle Hangmuren WMTS
  SUBDATASET_4_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/wmts/dzsf/WMTSCapabilities.xml?,layer=potentielle_murgaenge_wmts
  SUBDATASET_4_DESC=Layer potentielle Murg├ñnge WMTS
  SUBDATASET_5_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/wmts/dzsf/WMTSCapabilities.xml?,layer=potentielle_murgaenge_mangmuren_wmts
  SUBDATASET_5_DESC=Layer potentielle Murg├ñnge und Hangmuren WMTS
  SUBDATASET_6_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/wmts/dzsf/WMTSCapabilities.xml?,layer=hangmurengefaehrdung_wmts
  SUBDATASET_6_DESC=Layer Hangmurengef├ñhrdung WMTS
  SUBDATASET_7_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/wmts/dzsf/WMTSCapabilities.xml?,layer=murganggefaehrdung_wmts
  SUBDATASET_7_DESC=Layer Murganggef├ñhrdung WMTS
  SUBDATASET_8_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/wmts/dzsf/WMTSCapabilities.xml?,layer=mur_und_hangmur_wmts
  SUBDATASET_8_DESC=Layer Murgang- und Hangmurengef├ñhrdung WMTS
  SUBDATASET_9_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/wmts/dzsf/WMTSCapabilities.xml?,layer=boeschungsbrandgefaehrdung_agg_wmts
  SUBDATASET_9_DESC=Layer (Beta) B├Âschungsbrandgef├ñhrdung aggregiert WMTS
  SUBDATASET_10_NAME=WMTS:https://geoinformation.eisenbahn-bundesamt.de/wmts/dzsf/WMTSCapabilities.xml?,layer=boeschungsbrandgefaehrdung_wmts
  SUBDATASET_10_DESC=Layer (Beta) B├Âschungsbrandgef├ñhrdung WMTS
```
And the second one is correct.

### Versions and provenance

Windows 11, GDAL 3.9.2, GDAL bin from conda-forge

**Comment by jratike80** [comment]
The C API seems to search different metadata from the WMTS capabilities, the "ows:Metadata" thing.

```
<ows:Title>Hochwassergefährdung_HQ100 WMTS</ows:Title>
<ows:Identifier>hochwassergefaehrdung_wmts</ows:Identifier>
<ows:Metadata xlink:href="https://geoinformation.eisenbahn-bundesamt.de/geonetwork/dzsf/ger/csw?SERVICE=CSW&Version=2.0.2&REQUEST=GetRecordById&elementSetName=full&OutputSchema=http://www.isotc211.org/2005/gmd&ID=d8034bfc-b7ec-4f19-bbe0-63f61b9a31e8"/>
```

**Comment by michkowalczuk** [comment]
> The C API seems to search different metadata from the WMTS capabilities https://geoinformation.eisenbahn-bundesamt.de/wmts/dzsf/WMTSCapabilities.xml, the "ows:Metadata" thing.
> 
> ```
> <ows:Title>Hochwassergefährdung_HQ100 WMTS</ows:Title>
> <ows:Identifier>hochwassergefaehrdung_wmts</ows:Identifier>
> <ows:Metadata xlink:href="https://geoinformation.eisenbahn-bundesamt.de/geonetwork/dzsf/ger/csw?SERVICE=CSW&Version=2.0.2&REQUEST=GetRecordById&elementSetName=full&OutputSchema=http://www.isotc211.org/2005/gmd&ID=d8034bfc-b7ec-4f19-bbe0-63f61b9a31e8"/>
> ```

Shouldn't they both return the same to properly assign the selected subdataset as a new dataset using proper SUBDATASET_X_NAME?

**Comment by rouault** [comment]
I can replicate this by doing gdalinfo "https://geoinformation.eisenbahn-bundesamt.de/wmts/dzsf/WMTSCapabilities.xml", without the WMTS prefix. The issue must be that you do the same thing in your code

I've also queue a fix in https://github.com/OSGeo/gdal/pull/10733 so we do the right thing in that situation

**Comment by michkowalczuk** [comment]
> I can replicate this by doing gdalinfo "https://geoinformation.eisenbahn-bundesamt.de/wmts/dzsf/WMTSCapabilities.xml", without the WMTS prefix. The issue must be that you do the same thing in your code
> 
> I've also queue a fix in #10733 so we do the right thing in that situation

You had right - my dataset path has no `WMTS:` prefix. Should I close this issue?
